Bindings/Java/Matlab/Utilities/osimTableToStruct.m

```diff
@@ -77,14 +77,19 @@
     
     % Get the osim table column label
     col_label  = char(osimtable.getColumnLabels.get(iLabel));
-    
+    disp_label = col_label;
     % MATLAB structs must start with a letter, and can only contain
     % letters, digits, and underscores.
-    % Remove an initial slash.
-    col_label = regexprep(col_label, '^/', '');
-    % Replace '/' and '|' if they are present.
-    col_label = strrep(col_label,'/', '_');
-    col_label = strrep(col_label,'|', '_');
+    illegalChars = [{' '} {'*'} {'.'} {'/'} {'-'} {'|'}];
+    for i = 1 : length(illegalChars)
+        if ~isempty(strfind(col_label, illegalChars{i}))
+           col_label = strrep(col_label,illegalChars{i}, '_'); 
+        end
+    end
+    % If the label has changed, inform the User.
+    if ~strcmp(disp_label,col_label)
+        disp(['Illegal Column label. ' disp_label ' changed to ' col_label ]); 
+    end
```

Comment:
Instead of a boolean, I just check to see if the label has changed after the for loop— does that sound okay?

Reply:
Ah yeah that works.